common/flatpak-transaction.c

```diff
@@ -2998,6 +3102,16 @@ flatpak_transaction_real_run (FlatpakTransaction *self,
   if (!ready_res)
     return flatpak_fail_error (error, FLATPAK_ERROR_ABORTED, _("Aborted by user"));
 
+#ifdef ENABLE_DISKSPACE_CHECK
+  flatpak_transaction_set_enable_diskspace_check (self, TRUE);
+#endif
+
+  if (priv->enable_diskspace_check)
+    {
+      if (!check_for_disk_space (self))
+        return flatpak_fail_error (error, FLATPAK_ERROR_ABORTED, _("Disk space checked failed"));
```

Comment:
I think we'd want a different error code so that UIs can distinguish this error from others. They may want to handle it specially – for example, an app store might suggest some large apps that could be removed.

The message is not very useful for a human, either. In what way did this fail? Don't we want something more like (not literally this message):

> "%s is needed to install/update these apps, but only %s is available", g_format_size (space_required), g_format_size (free_disk_space)

Reply:
I would be comfortable to show these sizes in the UI provided if the sizes are accurate.
Maybe we can pipe the sizes' down to `g_debug` and keep similar error message (but a more verbose than this) ?